lib/tapioca/compilers/dsl/active_record_scope.rb

```diff
@@ -0,0 +1,100 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "active_record"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::ActiveRecordScope` decorates RBI files for subclasses of
+      # `ActiveRecord::Base` which declare `scope` fields
+      # (see https://api.rubyonrails.org).
+      #
+      # For example, with the following `ActiveRecord::Base` subclass:
+      #
+      # ~~~rb
+      # class Post < ApplicationRecord
+      #   scope :public_kind, -> { where.not(kind: 'private') }
+      #   scope :private_kind, -> { where(kind: 'private') }
+      # end
+      # ~~~
+      #
+      # this generator will produce the RBI file `post.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # post.rbi
+      # # typed: true
+      # class Post
+      #   extend Post::GeneratedRelationMethods
+      # end
+      #
+      # module Post::GeneratedRelationMethods
+      #   sig { params(args: T.untyped, blk: T.untyped).returns(T.untyped) }
+      #   def private_kind(*args, &blk); end
+
+      #   sig { params(args: T.untyped, blk: T.untyped).returns(T.untyped) }
+      #   def public_kind(*args, &blk); end
+      # end
+      # ~~~
+      class ActiveRecordScope < Base
+        extend T::Sig
+
+        sig do
+          override.params(
+            root: Parlour::RbiGenerator::Namespace,
+            constant: T.class_of(::ActiveRecord::Base)
+          ).void
+        end
+        def decorate(root, constant)
+          scope_method_names = constant.send(:generated_relation_methods).instance_methods(false)
+          return if scope_method_names.blank?
```

Comment:
```suggestion
          return if scope_method_names.empty?
```

`Array#blank?` is an ActiveSupport extension and is not a part of Ruby. We already have `Array#empty?` in Ruby, and we are just interested in testing for the array being empty, so might as well use that.